adr/cli.py

```diff
@@ -15,129 +19,161 @@
-          }],
-    ],
-    'path': [
-        [['--path'],
-         {'required': True,
-          'help': "Path relative to repository root (file or directory)",
-          }],
-    ],
-    'platform': [
-        [['-p', '--platform'],
-         {'default': 'windows10-64',
-          'help': "Platform to limit results to (default: windows10-64)",
-          }],
-    ],
-    'rev': [
-        [['-r', '--revision'],
-         {'dest': 'rev',
-          'required': True,
-          'help': "Revision to limit results to",
-          }],
-    ],
-    'test': [
-        [['-t', '--test'],
-         {'required': True,
-          'dest': 'test_name',
-          'help': "Path to a test file",
-          }],
-    ],
-}
-"""
-These are commonly used arguments which can be re-used. They are shared to
-provide a consistent CLI across recipes.
-"""
-
-
-class RecipeParser(ArgumentParser):
-    arguments = []
-
-    def __init__(self, *groups, **kwargs):
-        ArgumentParser.__init__(self, **kwargs)
-
-        for cli, kwargs in self.arguments:
-            self.add_argument(*cli, **kwargs)
-
-        for name in groups:
-            group = self.add_argument_group("{} arguments".format(name))
-            arguments = ARGUMENT_GROUPS[name]
-            for cli, kwargs in arguments:
-                group.add_argument(*cli, **kwargs)
-
-
-def run_recipe(recipe, args, fmt='table'):
-    modname = '.recipes.{}'.format(recipe)
-    mod = importlib.import_module(modname, package='adr')
-    output = mod.run(args)
-
-    if isinstance(fmt, string_types):
-        fmt = all_formatters[fmt]
-
-    log.debug("Result:")
-    return fmt(output)
-
-
-def cli(args=sys.argv[1:]):
-    from adr.query import set_active_data_url
-    parser = ArgumentParser()
-    parser.add_argument('recipes', nargs='*', help="Recipe(s) to run.")
+    :param ArgumentParser parser: instance of an ArgumentParser.
+    :returns: ArgumentParser
+    """
+    # optional arguments
     parser.add_argument('-l', '--list', action='store_true', default=False,
                         help="List available recipes.")
     parser.add_argument('-f', '--format', dest='fmt', default='table',
                         choices=all_formatters.keys(),
                         help="Format to print data in, defaults to 'table'.")
     parser.add_argument('-v', '--verbose', action='store_true', default=False,
                         help="Print the query and other debugging information.")
-    parser.add_argument('--url', default='http://activedata.allizom.org/query',
+    parser.add_argument('-u', '--url', default='http://activedata.allizom.org/query',
                         help="Endpoint URL")
-    args, remainder = parser.parse_known_args(args)
-    set_active_data_url(args.url)
-    if args.verbose:
-        log.setLevel(logging.DEBUG)
-    else:
-        log.setLevel(logging.INFO)
+    # positional arguments
+    parser.add_argument('task', nargs='*', help='Task(s) to run.')
```

Comment:
I'd prefer not mentioning `task` in the help. I understand why you're calling it this (it could be a recipe or a query), let's just optimize the help text for the common case, i.e running recipes. The fact that it can also run queries can be an easter egg :)

Reply:
That's a fair point. I think it can be renamed back to `recipe` easily and would only require changes in a few places. Could be a super easy outreach task, even.